cdn/src/sentry.ts

```diff
@@ -0,0 +1,134 @@
+  if (typeof SENTRY_PROJECT_ID === 'undefined') return;
+  const body = JSON.stringify(toSentryEvent(err, request));
+
+  for (let i = 0; i <= RETRIES; i++) {
+    const res = await fetch(
+      `https://sentry.io/api/${SENTRY_PROJECT_ID}/store/`,
+      {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          "X-Sentry-Auth": [
+            "Sentry sentry_version=7",
+            `sentry_client=${CLIENT_NAME}/${CLIENT_VERSION}`,
+            `sentry_key=${SENTRY_KEY}`,
+          ].join(", "),
+        },
+        body,
+      }
+    );
+    if (res.status === 200) {
+      return;
+    }
+    // We couldn't send to Sentry, try to log the response at least
+    console.error({ httpStatus: res.status, ...(await res.json()) }); // eslint-disable-line no-console
+  }
+}
+
+export function toSentryEvent(err: Error, request: Request) {
+  const errType = err.name || ((err as any).contructor || {}).name;
+  const frames = parse(err);
+  const extraKeys = Object.keys(err).filter(
+    (key) => !["name", "message", "stack"].includes(key)
+  );
+  const { searchParams } = new URL(request.url);
+
+  const { v4 } = require('uuid-browser');
+  return { 
+    event_id: v4(),
+    message: errType + ": " + (err.message || "<no message>"),
+    exception: {
+      values: [
+        {
+          type: errType,
+          value: err.message,
+          stacktrace: frames.length ? { frames: frames.reverse() } : undefined,
+        },
+      ],
+    },
+    extra: extraKeys.length
+      ? {
+          [errType]: extraKeys.reduce(
+            (obj, key) => ({ ...obj, [key]: (err as any)[key] }),
+            {}
+          ),
+        }
+      : undefined,
+    tags: TAGS,
+    platform: "javascript",
+    environment: ENV,
+    server_name: SERVER_NAME,
+    timestamp: Date.now() / 1000,
+    request:
+      request && request.url
+        ? {
+            method: request.method,
+            url: request.url,
+            query_string: searchParams ? searchParams.toString() : '',
+            headers: request.headers,
+            data: request.body,
+          }
+        : undefined,
+    // release: SENTRY_RELEASE,
+  };
+}
+
+export function parse(err: Error) {
+  return (err.stack || "")
+    .split("\n")
+    .slice(1)
+    .map((line) => {
+      if (line.match(/^\s*[-]{4,}$/)) {
+        return { filename: line };
+      }
+
+      // From https://github.com/felixge/node-stack-trace/blob/1ec9ba43eece124526c273c917104b4226898932/lib/stack-trace.js#L42
+      const lineMatch = line.match(
+        /at (?:(.+)\s+\()?(?:(.+?):(\d+)(?::(\d+))?|([^)]+))\)?/
+      );
+      if (!lineMatch) {
+        return;
+      }
+
+      return {
+        function: lineMatch[1] || undefined,
+        filename: lineMatch[2] || undefined,
+        lineno: +lineMatch[3] || undefined,
+        colno: +lineMatch[4] || undefined,
+        in_app: lineMatch[5] !== "native" || undefined,
+      };
+    })
+    .filter(Boolean);
+}
```

Comment:
Not a strong ask at all, as it's largely unchanged implementation in years, but could we use [`stack-trace`](https://npm.im/stack-trace) here?  If the answer is _yes_, but we just don't, maybe just a comment suggesting that as an option (and a note that we've copied this for a particular reason, or that it was inspired by something).

Reply:
`stack-trace` won't run directly on workers :( This is a thin implementation that works on workers well